src/realm/object-store/c_api/object.cpp

```diff
@@ -28,6 +28,16 @@ RLM_API realm_object_t* realm_get_object(const realm_t* realm, realm_class_key_t
     });
 }
 
+RLM_API bool realm_object_get_parent(const realm_object_t* object, realm_object_t* parent)
+{
+    return wrap_err([&]() {
+        if (parent)
+            *parent = realm_object_t{realm::Object{object->get_realm(), object->obj().get_parent_object()}};
```

Comment:
I've seen the C API littered with these types of checks and I'm no sure I understand their purpose. Not supplying a pointer to store the parent in sounds like a fundamental error in using the API. I'd rather have this be an assert than the method silently not doing anything.

Reply:
I agree, but it is important we keep the same pattern, I can clean all this up in a separate PR.